src/python-common/ceph/deployment/service_spec.py

```diff
@@ -420,6 +421,52 @@ def from_string(cls, arg):
 _service_spec_from_json_validate = True
 
 
+class CustomConfig:
+    def __init__(self, content: str, mount_path: str) -> None:
+        self.content: str = content
+        self.mount_path: str = mount_path
+        self.validate()
+
+    def to_json(self) -> Dict[str, Any]:
+        return {
+            'content': self.content,
+            'mount_path': self.mount_path,
+        }
+
+    @classmethod
+    def from_json(cls, data: Dict[str, Any]) -> "CustomConfig":
+        for k in ['content', 'mount_path']:
+            if k not in data:
+                raise SpecValidationError(f'CustomConfig must have "{k}" field')
+        for k in data.keys():
+            if k not in ['content', 'mount_path']:
```

Comment:
it seems like these fields are mandatory `['content', 'mount_path']` .. maybe we should have them stored in a variable and use them in both for loops?

Reply:
added a `_fields` attribute to the `CustomConfig` class to hold these values